### DISCUSSION TITLE
Why Vitest does not typecheck with typescript?

### DISCUSSION THREAD

**Discussion by gal-yedidovich** [vitest-dev/vitest#3417] [Q&A]
with `vitest` the following test passes without errors:

```ts
import { it } from 'vitest'
it('example', () => {
	const a: number = 'string value'; // this is an invalid syntax
});
```

I would expect it to behave like `jest` and fail, with the error like `Type 'string' is not assignable to type 'number'.`

Is there a configuration to enable compiler check on the syntax & types? I would like to prevent mistakes from passing unintentionally

**Comment by jsphstls** [comment]
It is currently possible for `vitest` to typecheck all test files. However, the current documentation does not describe how to directly achieve this. Even worse, the documentation fails to disambiguate the term "typecheck." The docs largely refer to typechecking in the context of a _typecheck test_ where the test is performing typechecks instead of tests getting typechecked.

Until those issues are addressed, here's how to typecheck your tests with vitest.
1. Enable "typecheck" in the vitest config. See https://github.com/vitest-dev/vitest/issues/7773
2. Include at least one `*test-d.ts` file. See: https://github.com/vitest-dev/vitest/issues/5868

**Comment by Mythli** [comment]
This is super confusing
No logically thinking programmer would guess this.
